builds/jenkins-build-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,1062 @@
+      repocloneUrl = scmModule.getCoreRepoCloneUrl(repo_name)
+    } else {
+      auth_token = setCredentials()
+      repocloneUrl = scmModule.getRepoCloneUrl(repo_name)
+    }
+
+    dir(repo_name) {
+      checkout([$class: 'GitSCM', branches: [
+        [name: '*/' + params.scm_branch]
+      ], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [
+          [credentialsId: repo_credential_id, url: repocloneUrl]
+        ]])
+    }
+
+    def configObj = dir(repo_name) {
+      LoadConfiguration()
+    }
+
+    if (configObj.service_id) {
+      config = serviceMetadataLoader.loadServiceMetadata(configObj.service_id)
+    } else {
+      error "Service Id is not available."
+    }
+  }
+
+  if (!config) {
+    error "Failed to fetch service metadata from catalog"
+  }
+
+  dir(repo_name) {
+    scmModule.setServiceConfig(config)
+    def envApi = "${g_base_url}/jazz/environments"
+    environmentDeploymentMetadata.initialize(config, configLoader, scmModule, branch, env.BUILD_URL, env.BUILD_ID, envApi, auth_token)
+    gitCommitHash = scmModule.getRepoCommitHash()
+    gitCommitOwner = scmModule.getRepoCommitterInfo(gitCommitHash)
+    context_map = [created_by: config['created_by'], deployed_by: gitCommitOwner]
+    environment_logical_id = '';
+
+    if (branch == 'master') {
+      current_environment = 'prod'
+      environment_logical_id = 'prod';
+    } else {
+      current_environment = 'dev'
+      environment_logical_id = environmentDeploymentMetadata.getEnvironmentLogicalId();
+    }
+
+    if (!environment_logical_id && config['domain'] != 'jazz') {
+      error "The environment has not been created yet and its missing the environment id"
+    }
+  }
+  if (!events) {
+    error "Can't load events module"
+  } //Fail here
+  def eventsApi = "${g_base_url}/jazz/events"
+  events.initialize(configLoader, config, "SERVICE_DEPLOYMENT", branch, environment_logical_id, eventsApi)
+
+  def runtime = config['providerRuntime']
+  def service = config['service']
+  def isScheduleEnabled = isEnabled(config, "eventScheduleRate")
+  def isStreamEnabled = isEnabled(config, "event_source_kinesis")
+  def isDynamoDbEnabled = isEnabled(config, "event_source_dynamodb")
+  def isS3EventEnabled = isEnabled(config, "event_source_s3")
+  def isSQSEventEnabled = isEnabled(config, "event_source_sqs")
+  def isEc2EventEnabled = isEnabled(config, "event_source_ec2")
+  def isEventSchdld = false
+  def internalAccess = config['require_internal_access']
+  domain = config['domain']
+
+  def roleId = config['iamRoleARN'].substring(config['iamRoleARN'].indexOf("::") + 2, config['iamRoleARN'].lastIndexOf(":"))
+
+  sonarModule.initialize(configLoader, config, branch)
+  stackName = "${configLoader.INSTANCE_PREFIX}-${config['domain']}-${config['service']}-${environment_logical_id}"
+
+  if (isScheduleEnabled || isEc2EventEnabled || isS3EventEnabled || isStreamEnabled || isDynamoDbEnabled || isSQSEventEnabled){
```

Comment:
Not required again.

Reply:
Removing stackName variable and substituting it with the sequence of lamdba functions retrieved from the deployment_descriptor

Reply:
Resolved with https://github.com/tmobile/jazz/pull/829/commits/f334469d3950dbf33226188826b8897bbed9309b and https://github.com/tmobile/jazz/pull/829/commits/0ce1a577497d8248cc5fff9027c5fc4d9f7054cd However I did not touch sonarModule that seems still can be useful